### ISSUE TITLE
batteries and "charge elevator"

### ISSUE THREAD

**Issue by ericmazur** [ericmazur/PnPbook#61]
From @JohnDenker: The "charge elevator" shown in figure 31.2 is not a good
model of how a battery works.  It seems to be a recent
invention, popular in the "research based" textbooks, but
it's the sort of thing that gives "research" a bad name.

For one thing, according to the only way I can see to
interpret the diagram, it puts out both constant voltage
and constant current.  A real device that did that would
be extremely dangerous, for example if somebody accidentally
open-circuited it.  Real batteries are not remotely like 
that.

In any case, the "charge elevator" is unnecessary, to a
rather comical degree.  For 200 years or so, sensible 
people have been modeling batteries as high-value
capacitors.  That makes a fine model, good to first
order in charge and zeroth order in current.  It's
better than the "charge elevator" in every way.  If 
you add a little bit of series resistance, it becomes 
even better, i.e. good to first order in current.

**Comment by ericmazur** [comment]
Good point. I guess I do want to lay the 'ground work' for the "pipe-and-ball" model I use for junctions etc. I will rethink.

**Comment by JohnDenker** [comment]
However ... the pipe-and-ball model is predicated on some wrong physics.  A discussion of this point is in the email queue, and should appear on the tracker eventually.  In the meantime, you can watch the movie:
  https://www.av8n.com/physics/conservation-continuity.htm#sec-speedup

**Comment by ericmazur** [comment]
That's true, but so much is "wrong" if taken literally. It _does_ make the continuity point well.

**Comment by JohnDenker** [comment]
Still, though, the incompressible-balls model is completely unnecessary.  You can just say that (a) there are no sources and there are no sinks, because charge is conserved, and (b) in the steady state, charge isn't accumulating anywhere, therefore (c) what goes in must come out.  That's all that is needed for chapter 31.  Short and sweet and correct.

Anything students learn about incompressible balls will have to be unlearned almost immediately.  See e.g. the discussion of the Drude model later in chapter 31.  See also the discussion of semiconductors in section 32.3.  See also issue #66.

For chapter 32, you need (b') usually rather little stray charge is accumulating anywhere, because the stray capacitances are usually rather small relative to things you care about.  People like for circuits to behave this way, and generally design things accordingly.  However, this is an engineering goal, not a law of nature.  Beware that there _can_ be significant exceptions.  This is not a question of taking things over-literally.  For example:
  https://www.youtube.com/watch?v=9tzga6qAaBA&t=0m52s
